specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/workspaceRP.json

```diff
@@ -6353,8 +6825,15 @@
         "enableDataIsolation": {
           "type": "boolean"
         },
+        "enableServiceSideCMKEncryption": {
```

Comment:
>enableServiceSideCMKEncryption [](http://example.com/codeflow?start=9&length=30)

In place of having this enabling flags you can convert to an enum property which could have as values the types of features enabled like [ServiceSideCMKEncryption, SimplifiedCmk, SoftwareBillOfMaterials, DataIsolation]. In this way you dont need to add new properties in a new version

Reply:
I will sync with my team to create a property like you suggest in next version to avoid this in the future, thanks for the good suggestion.